builder_test.go

```diff
@@ -698,6 +698,23 @@ func TestBuilder(t *testing.T) {
 				Labels: map[string]string{"test": "value"},
 			},
 		},
+		{
+			Dockerfile: "dockerclient/testdata/Dockerfile.unset",
+			From:       "busybox",
+			Image: &docker.Image{
+				ID: "busybox2",
+				Config: &docker.Config{
+					Env: []string{},
+				},
+			},
+			ErrFn: func(err error) bool {
+				return err != nil && strings.Contains(err.Error(), "is not allowed to be unset")
+			},
```

Comment:
Is this test case expected to trigger this error?  In its current form, `TestBuilder()` can ignore an error, but it can't fail a test that's expecting to generate one.

Reply:
Yes testcase is expected to trigger the error.

When I alter 
```diff
- 				return err != nil && strings.Contains(err.Error(), "is not allowed to be unset")
+ 				return err != nil && strings.Contains(err.Error(), "is not allowed to be unset hello")
```
and run 
```console
make test
```

I get expected failure
```console
$ make test
go test ./...
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/README.md] -> /README.md (from: download:true), chown: , chmod 
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/LICENSE] -> / (from: download:true), chown: , chmod 
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/LICENSE] -> /A (from: download:true), chown: , chmod 
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/LICENSE] -> /a (from: download:true), chown: , chmod 
2023/02/03 21:10:35 RUN [mkdir ./b] [] true ([])
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/LICENSE] -> /b/a (from: download:true), chown: , chmod 
2023/02/03 21:10:35 COPY [https://github.com/openshift/origin/raw/master/LICENSE] -> /b/ (from: download:true), chown: , chmod 
2023/02/03 21:10:35 COPY [https://github.com/openshift/ruby-hello-world/archive/master.zip] -> /tmp/ (from: download:true), chown: , chmod 
--- FAIL: TestBuilder (0.00s)
    --- FAIL: TestBuilder/dockerclient/testdata/Dockerfile.unset_10 (0.00s)
        builder_test.go:845: 10: unexpected error: 10: 1: : resolve: Failed to process `${FOO:?}`: FOO is not allowed to be unset
2023/02/03 21:10:35 RUN [useradd -r -s /bin/false -m -d /home/${USER_NAME} ${USER_NAME}] [] true ([USER_NAME=my_user_env])
2023/02/03 21:10:35 ENSURE /home/my_user_env AS "my_user_env"
2023/02/03 21:10:35 RUN [echo $multival] [] true ([multivalarg=a=1 b=2 c=3 d=4 multival=a=1 b=2 c=3 d=4])
FAIL
FAIL	github.com/openshift/imagebuilder	0.007s
?   	github.com/openshift/imagebuilder/cmd/imagebuilder	[no test files]
ok  	github.com/openshift/imagebuilder/dockerclient	(cached)
?   	github.com/openshift/imagebuilder/dockerfile/command	[no test files]
ok  	github.com/openshift/imagebuilder/dockerfile/parser	(cached)
?   	github.com/openshift/imagebuilder/dockerfile/parser/dumper	[no test files]
ok  	github.com/openshift/imagebuilder/imageprogress	(cached)
?   	github.com/openshift/imagebuilder/signal	[no test files]
?   	github.com/openshift/imagebuilder/strslice	[no test files]
FAIL
make: *** [Makefile:6: test] Error 1
```

Therefore I assumed the `TestBuilder` is working correctly.

Reply:
Okay, after #244, if you set the `RunErr` field to the error checking callback, we should be able to be sure of that.

Reply:
Done.